bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/BootableJar.java

```
@@ -0,0 +1,371 @@
+        final LogContext embeddedLogContext = LogContext.create();
+        final Path bootLog = Paths.get(serverLogDir).resolve(SERVER_LOG);
+        final Path loggingProperties = Paths.get(serverCfgDir).resolve(Paths.get(LOGGING_PROPERTIES));
+        if (Files.exists(loggingProperties)) {
+            try (final InputStream in = Files.newInputStream(loggingProperties)) {
+                System.setProperty(LOG_BOOT_FILE_PROP, bootLog.toAbsolutePath().toString());
+                PropertyConfigurator configurator = new PropertyConfigurator(embeddedLogContext);
+                configurator.configure(in);
+            }
+        }
+        return embeddedLogContext;
+    }
+
+    public void run() throws Exception {
+        try {
+            server = buildServer(startServerArgs);
+        } catch (RuntimeException ex) {
+            cleanup();
+            throw ex;
+        }
+
+        Runtime.getRuntime().addShutdownHook(new ShutdownHook());
+        server.start();
+    }
+
+    private void cleanup() {
+        log.deletingHome(jbossHome);
+        deleteDir(jbossHome);
+
+    }
+
+    private Server buildServer(List<String> args) throws IOException {
+        String[] array = new String[args.size()];
+        log.advertiseOptions(args);
+        return Server.newSever(jbossHome, args.toArray(array), loader, this);
+    }
+
+    private static void deleteDir(Path root) {
+        if (root == null || !Files.exists(root)) {
+            return;
+        }
+        try {
+            Files.walkFileTree(root, new SimpleFileVisitor<Path>() {
+                @Override
+                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs)
+                        throws IOException {
+                    try {
+                        Files.delete(file);
+                    } catch (IOException ex) {
+                    }
```

Is there a reason we're swallowing the errors here?

---

We want to clean as much as we can even if a failure occurs (eg: Windows and opened jar deletion). I should add a log.